coverage-cop/action.yml

```diff
@@ -1,42 +1,123 @@
     required: false
-    default: 100
+    default: 95
   line-coverage-min:
     description: 'The minumum required line coverage (in %) for success'
     required: false
-    default: 100
+    default: 95
 runs:
   using: "composite"
-  steps: 
-      - name: Print coverage data
-        run: lcov --list --rc lcov_branch_coverage=1 ${{ inputs.path }}
+  steps:
+      - env:
+          stepName: Install Dependencies
+          bashPass: \033[32;1mPASSED -
+          bashInfo: \033[33;1mINFO -
+          bashFail: \033[31;1mFAILED -
+          bashEnd:  \033[0
         shell: bash
-      - name: Install lcov (if not present)
-        run: sudo apt-get install lcov
+        name: ${{ env.stepName }}
+        run: |
+          # ${{ env.stepName }}
+          echo "::group::${{ env.stepName }}"
+          sudo apt-get install lcov fd-find
+          echo "::endgroup::"
+          echo -e "${{ env.bashPass }} ${{env.stepName }} ${{ env.bashEnd }}"
+
+      - env:
+          stepName: Check Line and Branch Coverage
+          bashPass: \033[32;1mPASSED -
+          bashInfo: \033[33;1mINFO -
+          bashFail: \033[31;1mFAILED -
+          bashEnd:  \033[0
+        name: ${{ env.stepName }}
+        id: action-check-line-and-branch-coverage
         shell: bash
-      - name: Check coverage
         run: |
-          LINE_COVERAGE=$(lcov --list ${{ inputs.path }} | tail -n 1 | cut -d '|' -f 2 | sed -n "s/\([^%]*\)%.*/\1/p")
-          BRANCH_COVERAGE=$(lcov --rc lcov_branch_coverage=1 --list ${{ inputs.path }} | tail -n 1 | cut -d '|' -f 4 | sed -n "s/\([^%]*\)%.*/\1/p")
+          # ${{ env.stepName }}
+
+          # Print the received code cov report.
+          # TODO: The way it grabs the line/branch coverage is a little complicated
+          # I'd like to see if this can be done simpler.
+          echo -e " ${{ env.bashInfo }} Received LCov Report: ${{ inputs.line-coverage-min }} ${{ env.bashEnd}}"
+          lcov --list --rc lcov_branch_coverage=1 ${{ inputs.coverage-file }}
+          LINE_COVERAGE=$(lcov --list ${{ inputs.coverage-file }} | tail -n 1 | cut -d '|' -f 2 | sed -n "s/\([^%]*\)%.*/\1/p")
+          BRANCH_COVERAGE=$(lcov --rc lcov_branch_coverage=1 --list ${{ inputs.coverage-file }} | tail -n 1 | cut -d '|' -f 4 | sed -n "s/\([^%]*\)%.*/\1/p")
           RESULT=0
-          echo "Required line coverage: ${{ inputs.line-coverage-min }}"
-          echo "Line coverage:   $LINE_COVERAGE"
+
+          # Check Line Coverage
+          echo -e " ${{ env.bashInfo }} Required Line Coverage: ${{ inputs.line-coverage-min }} ${{ env.bashEnd}}"
+          echo -e " ${{ env.bashInfo }} Received Line Coverage: $LINE_COVERAGE ${{ env.bashEnd}}"
           if [[ $(echo "$LINE_COVERAGE < ${{ inputs.line-coverage-min }}" | bc) -ne 0 ]]; then
-            echo "Line Coverage is too low."
+            echo -e "${{ env.bashFail }} Line Coverage is too low. ${{ env.bashEnd }}"
```

Comment:
What is `bc` here?

Reply:
bc - arbitrary-precision decimal arithmetic language and calculator
 Looks like it's using that to determine if the less then is true
 I don't think this needs to be as complicated as it is.
 But trying to limit scope and not change anything else at this point